teps/0086-changing-the-way-result-parameters-are-stored.md

```diff
@@ -0,0 +1,444 @@
+runs or they may be responsible for monitoring the execution of runs,
+via CLI, dashboard or a monitoring system.
+
+Consider including folks that also work on CLI and dashboard.
+-->
+
+* Users must be able to refer to these result parameters exactly as they would know and still be able to reference in subsequent tasks (i.e. read access)
+
+### Performance (optional)
+
+<!--
+Consideration about performance.
+What impact does this change have on the start-up time and execution time
+of task and pipeline runs? What impact does it have on the resource footprint
+of Tekton controllers as well as task and pipeline runs?
+
+Consider which use cases are impacted by this change and what are their
+performance requirements.
+-->
+
+## Design Details
+
+<!--
+This section should contain enough information that the specifics of your
+change are understandable.  This may include API specs (though not always
+required) or even code snippets.  If there's any ambiguity about HOW your
+proposal will be implemented, this is the place to discuss them.
+
+If it's helpful to include workflow diagrams or any other related images,
+add them under "/teps/images/". It's upto the TEP author to choose the name
+of the file, but general guidance is to include at least TEP number in the
+file name, for example, "/teps/images/NNNN-workflow.jpg".
+-->
+
+### Result References
+
+Beyond size limits of specific TaskRuns’ results, the fundamental issue is that API objects in etcd are not suitable for storing as much data as users want to be able to report in results, and pass as parameters.
+
+In addition to disaggregating TaskRun statuses into PipelineRun statuses (which we should do anyway before v1), we should introduce some way to pass one TaskRun’s output results into another TaskRun’s input parameters by reference, instead of only by value as they are today.
+
+**Example:**
+
+```yaml
+tasks:
+- name: first-task
+  taskSpec: foo
+  results:
+  - name: out
+    byRef: true # <-- *new field*
+- name: second-task
+  taskSpec: bar
+  params:
+  - name: in
+    value: "$(tasks.first-task.results.out)"
+```
+
+
+**Implementation:**
+
+Result references require some code to take the contents of `/tekton/results/out` in the container environment, copy them elsewhere, and mint a unique reference. It also requires some code to take that reference, locate the contents whever they are, and make them available to other TaskRuns at `/tekton/params/in`.
+
+Both of these could be satisfied by the same component: a Tekton-internal sidecar, the *Result Collector Sidecar*. 
+
+**Sidecar API:**
+
+1. The controller is passed a flag, -sidecar-image=<image>
+2. This image is added as a container in every TaskRun Pod, with each steps’ results emptyDir volume attached as read–only, and params emptyDir volumes attached as read-write.
+3. Watch directory, and any time it sees a result file appear, it writes that content externally (GCS, datastore, cuneiform tablets, the moon)
```

Comment:
am i right that the sidecar based solution could be combined with some of the other alternatives below? i.e. the sidecar could be responsible for writing to an HTTP service, to a CRD, etc. might be helpful to make it clear which options could work with the sidecar approach